ext/bit/bit_details.hpp

```diff
@@ -959,6 +960,46 @@ constexpr T _mulx(T src0, T src1, T* hi, X...) noexcept
 }
 // -------------------------------------------------------------------------- //
 
+enum range_type {
+  LEFT_ALIGNED,
+  RIGHT_ALIGNED,
+  INTERNAL,
+  FULL,
+  POINT
+};
+
+template <class WrappedIter>
+range_type get_range_type(bit_iterator<WrappedIter> lhs, bit_iterator<WrappedIter> rhs) {
```

Comment:
We should change the name to imply that the queried range must be within a word.